lib/hyrax/hirmeos/work_factory.rb

```diff
@@ -5,12 +5,23 @@ class Hyrax::Hirmeos::WorkFactory
   def self.for(resource:)
     work = Hyrax::Hirmeos::Client::Work.new
     work.title = resource.title
-    work.uri = [{ uri: resource_url(resource), canonical: true }]
+    work.uri = [{ uri: resource_url(resource), canonical: true },
+                { uri: resource.id }]
     work.type = "other" # Need to map Hyrax work type to configured work types in HIRMEOS
+    file_urls = add_file_urls(resource)
+    work.uri << file_urls if file_urls.present?
+    work.uri.flatten!
     work
   end
 
   def self.resource_url(work)
     Rails.application.routes.url_helpers.polymorphic_url(work)
   end
+
+  def self.add_file_urls(work)
+    files = work.file_sets
+    return if files.blank?
+    links = files.map { |file| Hyrax::Engine.routes.url_helpers.download_url(id: file) }
+    links.each_slice(1).map { |link| Hash[[:uri].zip(link)] }
```

Comment:
What about this instead?  I think it is a more readable.
```suggestion
    links.map { |link| { uri: link } }
```

_I did go looking for a better method and I'm guessing there might still be a simpler approach but I only found more complicated solutions._

Reply:
I agree that there is probably a better solution, I like you suggestion a lot better, I was looking for something exactly like this